adding bcf2 - weighted bcf - files

cdelannoy opened the pull request:
Jared,

This code allows weights to be passed to the bcf() function. When a user includes weights (argument w), BCF fits a model y | x, w ~ N(f(x), σ^2 / w).
Other items we changed include consolidating the likelihood computation, and renaming the package bcf2 to make it easier to install and compare both versions.
Let us know if you would like us to make any changes or share any other materials.
Looking forward to your feedback.

Best,
Constance, Mariel, Lauren & Peter

Commit:
adding bcf2 - weighted bcf - files

TychonautVII committed:
pulled in scaling

TychonautVII committed:
updated the code for scaling

cdelannoy committed:
Merge pull request #1 from mathematica-mpr/scaling

Scaling

cdelannoy committed:
re-instated option to log iterations with verbose argument

TychonautVII committed:
removing compiled files from repo

TychonautVII committed:
more updates to the gitignore

TychonautVII committed:
Merge pull request #2 from mathematica-mpr/cleanup

Cleanup

cdelannoy committed:
added code to save trees and make predictions

TychonautVII committed:
initial commit of cleanup speedup code

TychonautVII committed:
more bug fixing on moving speedup over

TychonautVII committed:
more cleanup and adding some speedup related arguements

cdelannoy committed:
added code to save trees and make predictions

cdelannoy committed:
updating gitignore to not include vs code files

TychonautVII committed:
Merge pull request #4 from mathematica-mpr/speedup

Speedup

TychonautVII committed:
added some notes about things to do to the readme

TychonautVII commented on src/bcf_clean_overpar.cpp:
```diff
@@ -329,41 +456,197 @@ List bcfoverparRcppClean(NumericVector y_, NumericVector z_,
-      precs[k] = bscale1*bscale1/(sigma*sigma);
-    }
-    for(size_t k=ntrt;k<n;k++) {
-      precs[k] = bscale0*bscale0/(sigma*sigma);
```
Old precs calculation

TychonautVII committed:
update

cdelannoy committed:
successfully predicted taus, yhats still need some work (tau potentially too)

TychonautVII committed:
todo list

cdelannoy committed:
fixed merge conflicts. man/bcf.Rd file will need to be re-created when we build pkg

TychonautVII committed:
more cleanup of package

TychonautVII committed:
increased cleanness of rstudio checks, no errors, understood warnings

TychonautVII committed:
got rid of some of the warnings in rstudio's checks

TychonautVII committed:
expanded on the package with usethis, mostly setting up skeletons

TychonautVII committed:
skeleton of a vignette

TychonautVII committed:
removed the weights from the dinfo object

TychonautVII committed:
removing y2 from signature of lil

TychonautVII committed:
removed unnessisary y2 calcs

TychonautVII committed:
removed y2 from singo

TychonautVII committed:
made it so we can still compute y2 over all ys with new sinfo

TychonautVII committed:
updated readme

TychonautVII committed:
cleaned up the vinyette stuff

TychonautVII committed:
updated build ignore

cdelannoy committed:
added z pred argument

TychonautVII committed:
commit of updates required for a weighted draw of the scales

TychonautVII committed:
left out a semicolin

cdelannoy committed:
Merge pull request #8 from mathematica-mpr/scales_gibs_draw_update

Scales gibs draw update - looks good to me!

cdelannoy committed:
fixed tau_pred, now matches perfectly in testing. yhat_preds still need some work...

cdelannoy committed:
Merge branch 'wbcf' into predict/keeptrees

cdelannoy committed:
returning bscale and mscale with predictions for testing

cdelannoy committed:
tau and mu preds now return the right output. Yhats are still different

cdelannoy committed:
changes for predict

cdelannoy committed:
minor cosmetic change

cdelannoy committed:
adding test scripts into examples folder

TychonautVII committed:
I think I made some progress here!

TychonautVII committed:
Merge branch 'wbcf' into R-ification

TychonautVII committed:
Merge pull request #6 from mathematica-mpr/R-ification

R ification

TychonautVII committed:
got started on a seperate predict function

TychonautVII committed:
cleaned up from previous commit

TychonautVII committed:
a little more progress

TychonautVII committed:
random seed implemented

TychonautVII committed:
updated paramater description

TychonautVII committed:
basic version of predict working

TychonautVII committed:
progress on pulling some of the unnesisary predict stuff out of the C

TychonautVII committed:
appears to work for me

TychonautVII committed:
cleaned up docs

TychonautVII committed:
resolved the merge conflicts

TychonautVII committed:
left some garbage in the git ignore

cdelannoy committed:
implemented first steps of  parallelization with doParallel

cdelannoy committed:
Merge pull request #9 from mathematica-mpr/random_seed

random seed implemented

cdelannoy committed:
resolved conflicts btw seed and predict updates

cdelannoy committed:
Merge pull request #10 from mathematica-mpr/predict/seperate_func

Predict/seperate func

cdelannoy committed:
resolved merge conflicts with wbcf predict and seed branches

cdelannoy committed:
fixed predict conflicts

cdelannoy committed:
added parbcf function to package

TychonautVII committed:
updated bcf for multi-chain

TychonautVII committed:
updated bcf with more helper functions

TychonautVII committed:
updated the documentation

TychonautVII committed:
reasonably complete draft of the MCMC object multi-chain thing

TychonautVII committed:
updating bcf examples

TychonautVII committed:
inital progress getting the bscales out

TychonautVII committed:
more progress on diagnostics

TychonautVII committed:
updated formula for mu

TychonautVII committed:
corrected mu output

TychonautVII committed:
First draft of single-chain predict

TychonautVII committed:
resolved a percision issue with predict

TychonautVII committed:
setup for multi-chain predict

TychonautVII committed:
got a draft of multichain predict

TychonautVII committed:
got the parrellel stuff working again

TychonautVII committed:
dropped tau scale

TychonautVII committed:
removed some nonsense

TychonautVII committed:
removed misleading comment

TychonautVII committed:
setup Test for myself

TychonautVII committed:
refactored the get suffs for clarity

TychonautVII committed:
getSuffDeath Compiles sucessfully

TychonautVII committed:
more logic for get suff death fleshed out

TychonautVII committed:
first working draft of the death parrellization

TychonautVII committed:
updated example

TychonautVII committed:
Merge pull request #12 from mathematica-mpr/getsuffDeathSpeedup

Getsuff death speedup

cdelannoy committed:
Merge pull request #11 from mathematica-mpr/chains/parallel_pbm

Chains/parallel pbm

cdelannoy committed:
updated example with parallel structure

cdelannoy committed:
Merge pull request #13 from mathematica-mpr/chains/parallel

Chains/parallel to wbcf.

cdelannoy committed:
updated package dependencies

cdelannoy committed:
added documentation start

idest88 committed:
LVF edits

cdelannoy committed:
updated website and function descriptions

idest88 committed:
added text, structure

idest88 committed:
corrected citation

cdelannoy committed:
removed pipe for examples

idest88 committed:
expanded updates

idest88 committed:
added text

cdelannoy committed:
added site html files and fixed bug in saving tree files

marielfinucane committed:
README edits and comments

I made a few small inline edits and marked comments with "MMF"

marielfinucane committed:
Little tweaks to inline edits

marielfinucane committed:
updating hyperlink

marielfinucane committed:
one last tweak to README!

marielfinucane committed:
Comments/edits on first sections of A Simple Example

... through the end of the "Fit BCF model" section

marielfinucane committed:
more simple_example edits/comments

cdelannoy committed:
fixed html with equation and incorporated Mariel's suggestions in README

cdelannoy committed:
Merge branch 'wbcf' of https://github.com/mathematica-mpr/bcf-1 into wbcf

cdelannoy committed:
added weights to variance of y

TychonautVII committed:
very basic cleanup to get the code to run

TychonautVII committed:
basic updates to summary

TychonautVII committed:
added in some commed out functionality for named paramaters

marielfinucane committed:
Suggested updates to the website's "Reference" tab

marielfinucane committed:
Suggested changes to predict-example flagged with "MMF"

marielfinucane committed:
small inline edits to NEWS.md; comments marked "MMF"

cdelannoy committed:
made slight edits to wording

cdelannoy committed:
documented changes with devtools

cdelannoy committed:
fixed merge conflicts

cdelannoy committed:
bug fix in predict function

TychonautVII committed:
fixed bug in GR printing

TychonautVII committed:
more suggested cleanup

TychonautVII committed:
tiny editoral edits

TychonautVII committed:
Merge pull request #15 from mathematica-mpr/pbm_review_2019_12_23

Pbm review 2019 12 23

TychonautVII committed:
updates to the news package

TychonautVII committed:
updated the function documentation

TychonautVII committed:
updated bcf

TychonautVII committed:
through most of the predict examples

TychonautVII committed:
updates to the simple example

TychonautVII committed:
updated doc with edits

TychonautVII committed:
updated the documents

TychonautVII committed:
updated BCF package

TychonautVII committed:
Merge pull request #16 from mathematica-mpr/features/2019_12_24_mf_edits

Features/2019 12 24 mf edits

idest88 committed:
lvf commit for pull (?)

idest88 committed:
pull issue (?)

Merge branch 'wbcf' of github.com:mathematica-mpr/bcf-1 into wbcf

# Conflicts:
#	src/RcppExports.o
#	src/bcf2.dll
#	src/bcf_clean_overpar.o
#	src/bd.o
#	src/funs.o
#	src/logging.o
#	src/rng.o
#	src/tree.o

idest88 committed:
text edits

cdelannoy committed:
added text and figures based on J and J's comments
